app/models/loan_response.rb

```diff
@@ -38,7 +40,14 @@ def hash_data
   end
 
   def embeddable_media
-    embeddable_media_id.present? ? EmbeddableMedia.find(embeddable_media_id) : nil
+    if embeddable_media_id.present?
+      result = EmbeddableMedia.find_safe(embeddable_media_id)
+    end
+    unless result
+      result = EmbeddableMedia.where(owner_id: @owner.id, owner_type: @owner.class.name,
+        owner_attribute: @custom_field.id).first
+    end
+    result
```

Comment:
Sounds good to me.

On Thu, Aug 4, 2016 at 10:52 AM, Joseph Elliott <notifications@github.com>
wrote:

> In app/models/loan_response.rb
> <https://github.com/sassafrastech/madeline_system/pull/117#discussion_r73537306>
> :
>
> > @@ -38,7 +40,14 @@ def hash_data
> >    end
> >
> >    def embeddable_media
> > -    embeddable_media_id.present? ? EmbeddableMedia.find(embeddable_media_id) : nil
> > +    if embeddable_media_id.present?
> > +      result = EmbeddableMedia.find_safe(embeddable_media_id)
> > +    end
> > +    unless result
> > +      result = EmbeddableMedia.where(owner_id: @owner.id, owner_type: @owner.class.name,
> > +        owner_attribute: @custom_field.id).first
> > +    end
> > +    result
>
> Given that the migration has now been optimized to run considerably
> faster, it's probably reasonable to remove the adaptive logic and simply
> leave the spreadsheet data missing for those with an old database. Agreed?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/117/files/64f1f13021e4e4a548f0c38083cc75756f92428d#r73537306>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq3cNvDuuEfPRa-PZm3c4n5XK5T7raGks5qcfymgaJpZM4JSkza>
> .
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org